deposit_contract.sol

```diff
@@ -26,16 +26,14 @@ interface IDepositContract {
 contract DepositContract is IDepositContract {
     uint constant GWEI = 1e9;
 
-    uint constant MIN_DEPOSIT_AMOUNT = 1000000000; // Gwei
     uint constant DEPOSIT_CONTRACT_TREE_DEPTH = 32;
-    uint constant MAX_DEPOSIT_COUNT = 4294967295; // 2**DEPOSIT_CONTRACT_TREE_DEPTH - 1
+    uint constant MAX_DEPOSIT_COUNT = 2**DEPOSIT_CONTRACT_TREE_DEPTH - 1;
     uint constant PUBKEY_LENGTH = 48; // bytes
     uint constant WITHDRAWAL_CREDENTIALS_LENGTH = 32; // bytes
     uint constant SIGNATURE_LENGTH = 96; // bytes
-    uint constant AMOUNT_LENGTH = 8; // bytes
 
     bytes32[DEPOSIT_CONTRACT_TREE_DEPTH] branch;
-    uint256 deposit_count;
+    uint64 deposit_count;
```

Comment:
The reason I kept `uint256` to try keeping the storage layout the same. Happy if we can move on though. Btw, the shorter size increases gas costs.

Reply:
FWIW the storage layout is already different, since vyper stores fixed size arrays the same way solidity stores dynamically sized arrays; `keccak256(slot) + index`, while Solidity stores it at `slot + index`. 

I think the gas-readability trade off is worth it!